src/cpp/core/include/core/json/Json.hpp

```diff
@@ -84,6 +131,46 @@ bool fillVectorString(const Array& array, std::vector<std::string>* pVector);
 bool fillVectorInt(const Array& array, std::vector<int>* pVector);
 bool fillMap(const Object& array, std::map< std::string, std::vector<std::string> >* pMap);
 
+namespace detail {
+
+inline int intField(const Object& object,
+                    const std::string& name,
+                    int ifNotFound,
+                    const ErrorLocation& location)
+{
+   if (object.count(name) == 0)
+   {
+      log::logWarningMessage("No field named '" + name + "'", location);
+      return ifNotFound;
+   }
+   
+   return const_cast<Object&>(object)[name].get_int();
+}
+
+inline std::string stringField(const Object& object,
+                               const std::string& name,
+                               const std::string& ifNotFound,
+                               const ErrorLocation& location)
+{
+   if (object.count(name) == 0)
+   {
+      log::logWarningMessage("No field named '" + name + "'", location);
+      return ifNotFound;
+   }
+   
+   return const_cast<Object&>(object)[name].get_str();
+}
+
+} // namespace detail
+
+#define JSON_STRING_FIELD(__OBJECT__, __NAME__, __IF_NOT_FOUND__)              \
```

Comment:
Sounds good to me -- I do personally like using macros for hiding this kind of stuff but I'll try to tone it done for code like this.